src/app.js

```diff
@@ -144,41 +75,7 @@ app.use((req, res, next) => {
  */
 redirects.setup(app);
 
-
-/* Charon API used by Auspice.
- */
-if (!PRODUCTION) {
-  // allow cross-origin from the gatsby dev server
-  app.use("/charon", cors({origin: 'http://localhost:8000'}));
-}
-
-app.routeAsync("/charon/getAvailable")
-  .getAsync(endpoints.charon.getAvailable);
-
-app.routeAsync("/charon/getDataset")
-  .getAsync(
-    endpoints.charon.setSourceFromPrefix,
-    endpoints.charon.setDatasetFromPrefix,
-    endpoints.charon.canonicalizeDatasetPrefix,
-    endpoints.charon.getDataset);
-
-app.routeAsync("/charon/getNarrative")
-  .getAsync(
-    endpoints.charon.setSourceFromPrefix,
-    endpoints.charon.setNarrativeFromPrefix,
-    endpoints.charon.getNarrative);
-
-app.routeAsync("/charon/getSourceInfo")
-  .getAsync(
-    endpoints.charon.setSourceFromPrefix,
-    endpoints.charon.getSourceInfo);
-
-app.routeAsync("/charon/*")
-  .all((req) => {
-    utils.warn(`(${req.method}) ${req.url} has not been handled / has no handler`);
-    throw new NotFound();
-  });
-
+setupCharon(app);
```

Comment:
Forgot to mention that I added the section comments.

As for general organization, I changed it to be consistent with existing patterns in 2a92dda08fdf1411b74fc97d4df1bd992ef26a72...5acacf7b1a6144edc313fa1261a16db719d92d4e. The noticeable struggle here is that there are existing `groups.js` and `fetch.js` files but I think the routes setup code should be separate from those, so I put them in less-ideally-named files `groupsRoutes.js` and `fetchRoutes.js`. What do you think?

Reply:
Agreed organization is important, but ah, sorry, I didn't mean to suggest you had to shift things to `module.setup()` vs. `setupModule()`! just wanted to note the discordance in the conversation about naming.

I'm a proponent of shared suffixes/prefixes/etc. for collections of related files like these routes/app setup functions, so names like `groupsSetup.js` and `groupsRoutes.js` make sense to me, especially in the context of _everything_ being `…Setup.js` or `…Routes.js`. I'd personally probably head for `src/routes/groups.js` (or `src/appSetup/groups.js`), paralleling `src/endpoints/…`.

I think clearly grouping together the routes/app setup files is quite useful, compared to spreading them over the top-level namespace. That would also sort of partially preserve the now-defunct principle motivating the previous organization, "The longer we can keep all (server-side) routes in one place, the better."

Reply:
Hmm, in particular, I don't think 5acacf7b1a6144edc313fa1261a16db719d92d4e is desirable. Since they're not being re-used by multiple apps, those reorganized routes serve only to make it harder to see the full routes registered with the app.

Reply:
> Hmm, in particular, I don't think 5acacf7b1a6144edc313fa1261a16db719d92d4e is desirable. Since they're not being re-used by multiple apps, those reorganized routes serve only to make it harder to see the full routes registered with the app.

This is where we have different perspectives, and I think it's due to our different levels of familiarity with the codebase. To me, the re-organization is incomplete without that change. The ability to see the full routes registered with the app is already impaired by defining shared routes elsewhere. This is also why I've removed the principle entirely - I don't see the point in preserving it if not followed by half the routes.

I agree the current implementation which spreads files over the top-level namespace needs a replacement. I'll propose these options - let me know what sounds good to you or if you have other ideas.

1. Use `src/routes/…`. I'm in favor of this if all routes are defined within the directory, including 5acacf7b1a6144edc313fa1261a16db719d92d4e and authentication routes currently defined in `src/authn/index.js`. This would fully preserve the principle.
2. Use `src/appSetup/…` without 5acacf7b1a6144edc313fa1261a16db719d92d4e. I'd update `app.js` with comments that include all imported routes (similar to [what's currently done for authentication](https://github.com/nextstrain/nextstrain.org/blob/7ae36f5427d6b21cec3b8f110fccaacc44ed5503/src/app.js#L108-L111)). It would mildly bug me that not all "app setup" is within this directory, but I think fine given the usage context would make that clear.

Reply:
Either of those options would be fine, I think.

> The ability to see the full routes registered with the app is already impaired by defining shared routes elsewhere. … I don't see the point in preserving it if not followed by half the routes.

The point is to not throw the baby out with the bathwater. Being able to see some decent chunk of routes all together, in order, in the same file is still useful and still reduces cognitive load, even if not all routes are there. It's that much less to look up when reading thru the app entrypoint. The point of the principle isn't consistency for consistency's sake; it's reducing cognitive load.

> This would fully preserve the principle.

Eh, not really? It brings the organization back a little closer to it, but the routes are still spread out across files and the ordering of them is still unclear without cross-referencing across files.

Reply:
Ok, I'll plan to do (1) + add comments to the entrypoint files to list out the imported routes.

Reply:
Done: 2a92dda08fdf1411b74fc97d4df1bd992ef26a72...8fa940c78e3363bae1550b3c00b00705dce93b8d

I went with `src/routing/` to broaden the scope to "anything that's related to routing", rather than "routes only", so files like `src/routing/setup.js` would not seem out of place. Maybe `src/express/` would be better? I can't tell...

Reply:
I think `src/routing/…` is fine. It's descriptive of most of the contents, and even if a few things in there are neither routes nor routing per se, it's all involved in how requests are routed to endpoints.

(And we can push things around later if, in time, we find it unsuitable.)